app/models/user.rb

```diff
@@ -94,4 +97,25 @@ def subscription?
   def email_verified?
     self.email && self.email !~ TEMP_EMAIL_REGEX
   end
+
+  def update_email_info!(new_email, subscribe)
```

Comment:
This is an instance method on user so we shouldn't need to pass these in. Rails gives you some nice `dirty` methods

```ruby
email_changed?
email_was
subscribed_changed?
subscribed_was
```

etc. This is a good place to use ActiveModel callbacks, like an `after_save` that checks whether we should subscribe or unsubscribe an email address for a user. We only perform actions if either of those values have changed.